protocols/1563-farmlab/nucleic_acid_purification.ot2.py

```diff
@@ -0,0 +1,153 @@
+            magdeck.disengage()
+
+
+def run_custom_protocol(number_of_sample_columns: int = 12,
+                        minutes_to_incubate_on_magnetic_block: int = 5,
+                        number_of_initial_mixes: int = 5,
+                        number_of_secondary_mixes: int = 25,
+                        elution_volume_ul: float = 100,
+                        number_of_first_supernatant_transfers: int = 6,
+                        number_of_second_supernatant_transfers: int = 2):
+    # setup samples
+    deep_samples = deep_plate.rows('A')[0:number_of_sample_columns]
+    mag_samples = mag_plate.rows('A')[0:number_of_sample_columns]
+    new_samples = new_plate.rows('A')[0:number_of_sample_columns]
+
+    for _ in range(5):
+        # mix and transfer samples to magnetic plate
+        for tip, source, dest in zip(sample_tips, deep_samples, mag_samples):
+            m300.pick_up_tip(tip)
+            m300.mix(number_of_initial_mixes, 150, source)
+            m300.transfer(180, source, dest.top(), new_tip='never')
+            m300.return_tip()
+
+        # incubate on engaged magnet for selected number of minutes
+        magdeck.engage()
+        m300.delay(minutes=minutes_to_incubate_on_magnetic_block)
+
+        # remove supernatant
+        for tip, s in zip(sample_tips, mag_samples):
+            m300.pick_up_tip(tip)
+            m300.transfer(200, s, waste, new_tip='never')
+            m300.return_tip()
+        magdeck.disengage()
+
+        # buffer transfers
+        buffer_transfer(VHB,
+                        VHB_tips,
+                        number_of_first_supernatant_transfers,
+                        mag_samples,
+                        number_of_secondary_mixes,
+                        True)
+        buffer_transfer(SPM,
+                        SPM_tips,
+                        number_of_second_supernatant_transfers,
+                        mag_samples,
+                        number_of_secondary_mixes,
+                        True)
+        buffer_transfer(SPM,
+                        SPM_tips,
+                        1,
+                        mag_samples,
+                        number_of_secondary_mixes,
+                        False)
+
+        for s_tip, w_tip, s in zip(sample_tips, water_tips, mag_samples):
+            m300.pick_up_tip(w_tip)
+            m300.transfer(180, water, s.top(), new_tip='never')
+            m300.drop_tip()
+
+            m300.pick_up_tip(s_tip)
+            m300.transfer(200, s, waste, new_tip='never')
+            m300.return_tip()
+        magdeck.disengage()
+
+        for tip, s in zip(elution_buffer_tips, mag_samples):
+            m300.pick_up_tip(tip)
+            m300.transfer(elution_volume_ul,
+                          elution_buffer,
+                          s.top(),
+                          new_tip='never')
+            m300.mix(number_of_secondary_mixes, 300, s)
+            m300.return_tip()
+
+        m300.delay(minutes=5)
```

Comment:
You might need to add `robot._driver.run_flag.wait()` before engaging the command, because the `pieptte.delay()` is not a module blocker. Please test on robot